addons/account_edi_ubl_cii/models/account_edi_common.py

```diff
@@ -391,6 +391,117 @@ def _invoice_constraints_common(self, invoice):
+                            'some_other_subtitle': {
+                                '_title': "Some Other Subtitle",
+                                'error_2': "Something is wrong with the other thing",
+                            },
+                        },
+                    }
+
+        Returns:
+            dict: Flattened dict where keys are group identifiers and values are formatted strings
+                  with titles and bullet-pointed messages, properly indented for nested groups.
+        """
+        def flatten_dict(d, level=0):
+            if not d:
+                return None
+
+            if all(k.startswith('_') for k in d):
+                return None
+
+            if '_title' not in d:
+                raise UserError(_("Missing '_title' for multi-level constraints."))
+
+            title = d.pop('_title')
+            simple_strings = []
+            children_strings = []
+
+            for k, v in d.items():
+                if v and isinstance(v, dict) and (flattened := flatten_dict(v, level + 1)):
+                    children_strings.append(flattened)
+                elif v and isinstance(v, str):
+                    simple_strings.append(v)
+
+            indent = '\t' * (level + 1) + indent_suffix
+            strings = [f'{'\t' * level}{indent_suffix}{title}']
+            for string in simple_strings:
+                strings.append(f'{indent}{string}')
+
+            for string in children_strings:
+                strings.append(string)
+
+            return '\n'.join(strings)
+
+        if not any(not k.startswith('_') and isinstance(v, dict) for k, v in constraints.items()):
+            # There are no dicts in relevant keys, so constraints is already flattened
+            return constraints
+
+        for k, v in constraints.items():
+            if not k.startswith('_') and not isinstance(v, (dict, str)):
+                # Only dict and str can be flattened
+                raise UserError(_("Constraints can not be flattened."))
+
+        residual_title = constraints.pop('_title', _("Other Errors:"))
+        residual_key = constraints.pop('_key', 'other')
+
+        if residual_key in constraints:
+            residuals = constraints[residual_key]
+        else:
+            residuals = dict()
+
+        # Aggregate residuals
+        for k, v in constraints.items():
+            if not k.startswith('_') and v and isinstance(v, str):
+                residuals[k] = v
+
+        # Remove residuals from root
+        for k in residuals:
+            del constraints[k]
+
+        if residuals:
+            residuals['_title'] = residual_title
+            constraints[residual_key] = residuals
+
+        flattened_constraints = dict()
+        for k, v in constraints.items():
+            if isinstance(v, dict):
+                if flattened := flatten_dict(v):
+                    flattened_constraints[k] = flattened
+            else:
+                flattened_constraints[k] = v
```

Comment:
```suggestion
        for k, v in constraints.items():
            flattened_constraints[k] = flatten(v)
```

On first glance this works, and removes the duplicate `isinstance(v, dict)` check that already exists in `flatten_dict`. Renaming it to `flatten` may be better in this case, because it also works on other types.